### PR TITLE
Pass mountPaths.appDirName to lifecycle as -app arg

### PR DESCRIPTION
**Pull request by jkutner**
## Summary

Fast follow fix for #1141

#### Before

app dir was mounted, but lifecycle was not `cd`-ing into it.

#### After

The provide workspace becomes the `CNB_APP_DIR`

## Documentation

- Should this change be documented?
    - [ ] Yes, see #___
    - [x] No

### CODE REVIEW

File: internal/build/lifecycle_execution.go [hunk 1]
```diff
@@ -193,7 +193,7 @@ func (l *LifecycleExecution) Create(ctx context.Context, publish bool, dockerHos
 
 	opts := []PhaseConfigProviderOperation{
 		WithFlags(l.withLogLevel(flags...)...),
-		WithArgs(repoName),
+		WithArgs("-app", l.mountPaths.appDirName(), repoName),
```
Comment (dfreilich) [inline internal/build/lifecycle_execution.go:196]: Does lifecycle expect it in the format `\appdir`, or in `c:\appdir`? If the later, we can use `mountPaths.appDir()` instead

**Review by dfreilich — COMMENTED** [review]
I should've thought of this when I merged in the other PR. Good catch!

File: internal/build/lifecycle_execution.go (hunk 1 above)
Comment (jkutner) [reply internal/build/lifecycle_execution.go:196]: oh, i might need to revert 51887a3

Comment (jkutner) [reply internal/build/lifecycle_execution.go]: fixed in 9cc6613

**Comment by dfreilich** [comment]
Closing this in favor of #1149

**Pull request closed without merging**